docker/utils/socket.py

```diff
@@ -64,5 +64,8 @@ def frames_iter(socket):
     """
     n = next_frame_size(socket)
     while n > 0:
```

Comment:
I feel like this could be rewritten in a more intuitive way. I feel like `n` should not be reset inside the loop that checks for its value. Here's what I would suggest as an alternative:

```python
while True:
  n = next_frame_size(socket)
  if n == 0:
    break
  while n > 0:
    result = read(socket, n)
    n -= len(result)
    yield result
```